pkgs/tools/inputmethods/skk/skk-dicts/default.nix

```diff
@@ -1,77 +1,192 @@
-    sha256 = "11cjrc8m99hj4xpl2nvzxanlswpapi92vmgk9d6yimdz0jidb6cq";
-  };
-  medium = fetchurl {
-    url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.M";
-    sha256 = "0pwjp9qjmn9sq6zc0k6632l7dc2dbjn45585ibckvvl9iwfqqxdp";
-  };
-  large = fetchurl {
-    url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.L";
-    sha256 = "0ps0a7sbkryd6hxvphq14i7g5wci4gvr0vraac8ia2ww67a2xbyc";
-  };
+  suffix = lib.optionalString useUtf8 ".utf8";
 
-  # english to japanese
-  edict = fetchurl {
-    url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.edict";
-    sha256 = "1vrwnq0vvjn61nijbln6wfinqg93802d2a8d4ad82n692v83b1li";
-  };
-  # misc
-  assoc = fetchurl {
-    url = "https://raw.githubusercontent.com/skk-dev/dict/8b35d07a7d2044d48b063d2774d9f9d00bb7cb48/SKK-JISYO.assoc";
-    sha256 = "1smcbyv6srrhnpl7ic9nqds9nz3g2dgqngmhzkrdlwmvcpvakp1v";
-  };
-in
-
-stdenv.mkDerivation {
-  pname = "skk-dicts-unstable";
-  version = "2020-03-24";
-  srcs = [ small medium large edict assoc ];
-  nativeBuildInputs = [ iconv skktools ];
+  mkDictNameValue =
+    { name
+    , description
+    , license # it's written in the beginning of each file
+    , files ? [ "SKK-JISYO.${name}" ]
+    }: {
+      name = lib.toLower (builtins.replaceStrings [ "." ] [ "_" ] name);
+      value = stdenvNoCC.mkDerivation {
+        pname = lib.toLower name;
+        version = "unstable-2023-02-07";
 
-  strictDeps = true;
+        src = fetchFromGitHub {
+          owner = "skk-dev";
+          repo = "dict";
+          rev = "1909dda026e6038975359a5eaeafcf50c9ce7fa3";
+          sha256 = "sha256-e7B1+ji/8ssM92RSTmlu8c7V5kbk93CxtUDzs26gE8s=";
+        };
 
-  dontUnpack = true;
+        nativeBuildInputs = lib.optionals useUtf8 [ nkf ];
 
-  installPhase = ''
-    function dictname() {
-      src=$1
-      name=$(basename $src)          # remove dir name
-      dict=$(echo $name | cut -b34-) # remove sha256 prefix
-      echo $dict
-    }
-    mkdir -p $out/share
+        buildPhase = ''
+          runHook preBuild
+        '' + lib.concatMapStrings
+          (file: ''
+            nkf -w ${file} \
+              | LC_ALL=C sed 's/coding: [^ ]\{1,\}/coding: utf-8/' \
+              > ${file + suffix}
+          '')
+          (lib.optionals useUtf8 (map lib.escapeShellArg files)) + ''
+          runHook postBuild
+        '';
 
-    for src in $srcs; do
-      dst=$out/share/$(dictname $src)
-      echo ";;; -*- coding: utf-8 -*-" > $dst  # libskk requires this on the first line
-      iconv -f EUC-JP -t UTF-8 $src | skkdic-expr2 >> $dst
-    done
+        installPhase = ''
+          runHook preInstall
+        '' + lib.concatMapStrings
+          (file: ''
+            install -Dm644 \
+              ${lib.escapeShellArg file} \
+              $out/share/skk/${lib.escapeShellArg (baseNameOf file)}
```

Comment:
Since this change increases the number of packaged dictionaries, I have moved the install path from `$out/share` to `$out/share/skk` for better organization. This path is also  adopted by Debian and RedHat.